modules/cactus-i18n/src/Components.tsx

```diff
@@ -114,31 +86,15 @@ const I18nSection: React.FC<I18nSectionProps> = ({
     const { controller, lang: ctxLang } = context
     lang = controller.negotiateLang(lang || ctxLang, true)[0]
     // Don't change the context until everything is loaded.
-    if (hasLoadedAll(controller, section, lang, dependencies)) {
+    if (hasLoadedAll(controller, section, lang)) {
```

Comment:
The `useI18nSection` hook gives you this as the return value, just move it up here:
```suggestion
    const isLoaded = useI18nSection({ section, lang, ...extraProps })
    if (isLoaded) {
```

Reply:
I was going to do this but this logic is inside an if/else block so I can't call the hook there. And I need to call `negotiateLang` before calling the hook, but to do that, I need the context to not be null. Not sure if there's a way to do what you're asking...